app/views/Project/Tagging/Sources/SourcesTable/EntryList/EntryItem/EntryControl/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useMemo, useContext } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import { IoCheckmark, IoClose } from 'react-icons/io5';
+import {
+    Button,
+} from '@the-deep/deep-ui';
+
+import {
+    MultiResponse,
+    Membership,
+} from '#types';
+
+import { EntryReviewComment } from '#types/newEntry';
+import { useModalState } from '#hooks/stateManagement';
+import { useLazyRequest, useRequest } from '#base/utils/restRequest';
+import UserContext from '#base/context/UserContext';
+
+import EntryUncontrolCommentModal from './EntryUncontrolCommentModal';
+import styles from './styles.css';
+
+interface Props {
+    className?: string;
+    entryId: number;
+    projectId: number;
+    value: boolean;
+    disabled?: boolean;
+    onChange: (entryId: number) => void;
+}
+
+interface EntryControlFormData {
+    commentType: number;
+}
+
+const CONTROL = 3;
+const QUALITY_CONTROLLER = 0;
+
+function EntryControl(props: Props) {
+    const {
+        className,
+        entryId,
+        projectId,
+        value,
+        onChange,
+        disabled,
+    } = props;
+
+    const {
+        user,
+    } = useContext(UserContext);
+
+    const {
+        pending: projectMembersPending,
+        response: projectMembersResponse,
+    } = useRequest<MultiResponse<Membership>>({
+        url: `server://v2/projects/${projectId}/project-memberships/`,
+        method: 'GET',
+        failureHeader: 'Project Membership',
+    });
+
+    const isQualityController = useMemo(() => (
+        projectMembersResponse?.results.find((v) => v.member.toString() === user?.id)
+            ?.badges.some((v) => (v === QUALITY_CONTROLLER))
+    ),
+    [projectMembersResponse?.results, user?.id]);
```

Comment:
@samshara , this should come from the ProjectContext.
@thenav56 , could you add `CAN_QUALITY_CONTROL` or similar permission in allowedPermissions?

Reply:
Sure. @AdityaKhatri .